### PR TITLE
Fix flaky widgets e2e tests by waiting for the snackbar correctly

### LINKED ISSUE #33275: Investigate unstable widgets test
## Description
This widgets test seems to be failing in trunk:
https://github.com/WordPress/gutenberg/runs/3014495507

```
FAIL specs/widgets/editing-widgets.test.js (114 s)
  ● Widgets screen › Should insert content using the inline inserter

    expect(received).toMatchInlineSnapshot(snapshot)

    Snapshot name: `Widgets screen Should insert content using the inline inserter 1`

    - Snapshot  - 9
    + Received  + 1

      Object {
    -   "sidebar-1": "<div class=\"widget widget_block widget_text\"><div class=\"widget-content\">
    - <p>First Paragraph</p>
    - </div></div>
    - <div class=\"widget widget_block\"><div class=\"widget-content\">
    - <h2>My Heading</h2>
    - </div></div>
    - <div class=\"widget widget_block widget_text\"><div class=\"widget-content\">
    - <p>Second Paragraph</p>
    - </div></div>",
    +   "wp_inactive_widgets": "",
      }

      374 | 		await saveWidgets();
      375 | 		const serializedWidgetAreas = await getSerializedWidgetAreas();
    > 376 | 		expect( serializedWidgetAreas ).toMatchInlineSnapshot( `
          | 		                                ^
      377 | 		Object {
      378 | 		  "sidebar-1": "<div class=\\"widget widget_block widget_text\\"><div class=\\"widget-content\\">
      379 | 		<p>First Paragraph</p>

      at Object.<anonymous> (specs/widgets/editing-widgets.test.js:376:35)
```

I couldn't see any artifacts for some reason.

### PR DESCRIPTION
**Pull request by kevin940726**
## Description

Related to #33275, but probably won't fix that entirely.

I think one reason behind the flaky e2e tests is that we are waiting for the wrong elements when the screen is being saved. We are waiting for elements with text of `Widgets saved.`, expecting it to be the snackbar element. But it turns out that there will always be a _a11y element_ that contains that text.

```html
<div id="a11y-speak-polite" class="a11y-speak-region" style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;" aria-live="polite" aria-relevant="additions text" aria-atomic="true">Widgets saved.&nbsp;</div>
```

Hence, the waiting resolves immediately, the widgets has not yet been saved. This is not always failing because we wait for another 500ms after saving, which is the root of the flakiness.

The solution is to correctly wait for the snackbar elements, and dismiss them once it's shown. We have to also enable **reduced motion** mode to disable the snackbar transition. (I think we should enable it for all of our e2e tests though, WDYT?)

## How has this been tested?

E2E tests should pass.

## Types of changes

Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/e2e-tests/specs/widgets/editing-widgets.test.js [hunk 1]
```diff
@@ -51,16 +51,23 @@ describe( 'Widgets screen', () => {
 	beforeAll( async () => {
 		// TODO: Ideally we can bundle our test theme directly in the repo.
 		await activateTheme( 'twentytwenty' );
-		await deactivatePlugin(
-			'gutenberg-test-plugin-disables-the-css-animations'
-		);
+		// Reduced motion is needed to immediately show and dismiss the snackbars.
+		await page.emulateMediaFeatures( [
```
Comment (kevin940726) [inline packages/e2e-tests/specs/widgets/editing-widgets.test.js:55]: We probably want to enable it globally too.

File: packages/e2e-tests/specs/widgets/editing-widgets.test.js [hunk 2]
```diff
@@ -51,16 +51,23 @@ describe( 'Widgets screen', () => {
...
+			{
+				name: 'prefers-reduced-motion',
```
Comment (youknowriad) [inline packages/e2e-tests/specs/widgets/editing-widgets.test.js:57]: A long time ago, I tried removing our `gutenberg-test-plugin-disables-the-css-animation` plugin and replace with this `page.emulateMediaFeatures`. I didn't manage to make it work at that time and it was not clear why. Maybe it's time to try that one more time :)

Comment (kevin940726) [reply packages/e2e-tests/specs/widgets/editing-widgets.test.js:57]: I believe only some of the animations/transitions are respecting `prefers-reduced-motion`. For instance, the transition effect in the customizer doesn't (https://core.trac.wordpress.org/ticket/53542). While some of the others don't respect the plugin (animations doing in JS like `react-spring`). Maybe we should enable them both?

Comment (youknowriad) [reply packages/e2e-tests/specs/widgets/editing-widgets.test.js:57]: the combination of the `gutenberg-test-plugin-disables-the-css-animation` and the `FORCE_REDUCED_MOTION` env variable that is enabled in e2e test builds disable all "Gutenberg animations" (the JS and the CSS ones), I do believe that ideally all animations should respect `prefers-reduced-motion` (including core ones) which might help us get rid of all these hacks (plugin and env variable)

Comment (kevin940726) [reply packages/e2e-tests/specs/widgets/editing-widgets.test.js:57]: `FORCE_REDUCED_MOTION` is only enabled on CI, and we have to manually rebuild the whole app if we're writing tests locally. Puppeteer's API seems like a better alternative here.

Yep, ideally, but we're not quite there yet 😅.

File: packages/e2e-tests/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### New Features
+
+-   Bail out tests if a prior snapshot failed. Fixed a bug which failing snapshots won't trigger screenshots [#33317](https://github.com/WordPress/gutenberg/pull/33317).
+
```
Comment (talldan) [inline packages/e2e-tests/CHANGELOG.md]: Seems unrelated, should it be in a separate PR?

File: packages/e2e-tests/specs/widgets/editing-widgets.test.js [hunk 4]
```diff
@@ -888,14 +895,18 @@ async function saveWidgets() {
...
+	await expect( savedSnackbarQuery ).toBeFound();
 
-	// FIXME: The snackbar above is enough for the widget areas to get saved,
-	// but not enough for the widgets to get saved.
-	// eslint-disable-next-line no-restricted-syntax
-	await page.waitForTimeout( 500 );
+	// Close the snackbar.
+	const savedSnackbar = await find( savedSnackbarQuery );
+	await savedSnackbar.click();
+	await expect( savedSnackbarQuery ).not.toBeFound();
```
Comment (talldan) [inline packages/e2e-tests/specs/widgets/editing-widgets.test.js:909]: Does it need to be dismissed for the test to work? If so, I think that's ok, but otherwise it should probably be in a separate test rather than something that happens any time widgets are saved.

**Review by talldan — APPROVED** [review]
LGTM, though I feel the snapshot thing should be in a separate PR.

I don't mind if you merge it though 😄

File: packages/e2e-tests/specs/widgets/editing-widgets.test.js (hunk 4 above)
Comment (kevin940726) [reply packages/e2e-tests/specs/widgets/editing-widgets.test.js:909]: They should be dismissed, or else there will multiple snackbars on the page, and we won't know when the widgets are saved. Which was the reason it was flaky before 😅

File: packages/e2e-tests/CHANGELOG.md (hunk 3 above)
Comment (kevin940726) [reply packages/e2e-tests/CHANGELOG.md]: I was just being lazy 😅, I can do that in another PR.

Comment (kevin940726) [reply packages/e2e-tests/CHANGELOG.md]: Moved it here: https://github.com/WordPress/gutenberg/pull/33448

File: packages/e2e-tests/specs/widgets/editing-widgets.test.js (hunk 4 above)
Comment (talldan) [reply packages/e2e-tests/specs/widgets/editing-widgets.test.js:909]: Ah, I see. If only the snackbar disappearing speed was proportional to the speed the e2e tests move. 😄 

Could consider adding a comment like 'avoids multiple snackbars being present on the screen'.
